### DISCUSSION TITLE
npx nuxi build cannot be completed since I enabled this "compressPublicAssets" option

### DISCUSSION THREAD

**Discussion by johan-Rm** [nuxt/nuxt#16169] [General]
------------------------------
ENVIRONMENT
------------------------------
- Operating System: `Linux`
- Node Version:     `v16.11.1`
- Nuxt Version:     `3.0.0`
- Nitro Version:    `1.0.0`
- Package Manager:  `yarn@1.22.19`
- Builder:          `vite`
- User Config:      `css`, `ssr`, `app`, `build`, `runtimeConfig`, `modules`, `typescript`, `devServerHandlers`, `plugins`, `intlify`, `content`, `experimental`, `components`, `nitro`, `routeRules`, `vite`, `image`, `tailwindcss`
- Runtime Modules:  `@vueuse/nuxt@9.6.0`, `@intlify/nuxt3@0.2.4`, `@nuxt/content@2.2.2`, `@pinia/nuxt@0.4.6`, `@nuxt/image-edge@1.0.0-27840416.dc1ed65`, `@nuxtjs/tailwindcss@6.1.3`, `nuxt-jsonld@2.0.7`, `~/modules/localeRoutes`
- Build Modules:    `-`
------------------------------

------------------------------
COMMANDS USED
------------------------------
`yarn build`
------------------------------

I'm about to deploy a project under Nuxt 3 (https://preprod.kazengarden.com), I'm trying to optimize the app, when I activated the "compressPublicAssets" option the build does not finish not?

In my public folder I have more than 300 images with subfolders
- public/images/*.jpg
- public/images/horizontal/*.jpg
- public/images/vertical/*.jpg

The process is frozen here :

yarn run v1.22.19
$ nuxt build
Nuxi 3.0.0
Nuxt 3.0.0 with Nitro 1.0.0
✔ Using ~/components/content for components in Markdown
ℹ Using Tailwind CSS from ~/assets/styles/tailwind.css
ℹ Client built in 20139ms
ℹ Building server...
✔ Server built in 26984ms

--------------------------------------
I don't receive any error and I don't know where to start to debug all this, can someone help me ?

**Comment by johan-Rm** [comment]
ok so i let the process run until the end it took 10 minutes, surely because of the images?

**Reply by nathanchase** [reply]
Yeah, you don't want to compress .JPGs with gzip/brotli. You really only want to compress these MIME types:

> text/html
> text/css
> text/plain
> text/xml
> text/x-component
> text/javascript
> application/x-javascript
> application/javascript
> application/json
> application/manifest+json
> application/vnd.api+json
> application/xml
> application/xhtml+xml
> application/rss+xml
> application/atom+xml
> application/vnd.ms-fontobject
> application/x-font-ttf
> application/x-font-opentype
> application/x-font-truetype
> image/svg+xml
> image/x-icon
> image/vnd.microsoft.icon
> font/ttf
> font/eot
> font/otf
> font/opentype 

I do see in https://github.com/unjs/nitro/blob/6541373545ee2716566bce46c7568ad2c3164cdf/src/compress.ts that `image` is included as a `isCompressableMime` 

@pi0 - maybe image should be removed from the mime types that should be compressable, since you don't want to double compress already compressed files (.jpg, .png, .etc.) - except the few in the above list...

**Reply by TheAlexLichter** [reply]
@nathanchase Sounds like a good candidate for a PR 👍

**Comment by dladeira** [comment]
I'm having the same problem. Adding `compressPublicAssets: true` makes it take forever. @pi0 - is there a fix for this, removing certain file types from mime types like @nathanchase said, or is this something we have to deal with?

I'm using nuxt `3.9.3`.

----

After more digging I figured out this is because I have a releases directory with a couple apps that were >50MB, and nitro was probably trying to compress them. I'm trying to ignore them, but if I set it to ignore in prerender, like this, it doesn't ignore the directory

```
prerender: {
    ignore: ["**/releases/**"],
}
```

But if I do it globally, then it doesn't copy the directory at all.

```
ignore: ["**/releases/**"],
```

Is there a way to ignore but only from compressing, and not from copying? Does this have something to do with me not being able to ignore in the prerender?
